### PR TITLE
Rename 3pc -> PArSEC

### PR DESCRIPTION
**Pull request by maurermi**
Renames instances of 3pc to PArSEC, includes scripts used for refactoring as well. Creating PR for review.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -76,20 +76,20 @@ COPY --from=builder /opt/tx-processor/build/src/uhs/client/client-cli ./build/sr
 # Copy atomizer config
 COPY --from=builder /opt/tx-processor/atomizer-compose.cfg ./atomizer-compose.cfg
 
-# Create 3PC Deployment Image
-FROM $IMAGE_VERSION AS threepc
+# Create PARSEC Deployment Image
```
Comment (AlexRamRam) [inline Dockerfile]: lower-case r for consistency?

File: README.md [hunk 2]
```diff
@@ -40,20 +40,20 @@ There are two UHS-based architectures as follows:
 
 Read the [2PC & Atomizer architecture guide](docs/uhs-architectures.md) for a detailed description of the system components and implementation of each architecture.
 
-## Generic Smart Contract Platform ("3PC")
-We built a system that is capable of performing parallel executions of generic smart contracts.
+## Parallel Architecture for Scalably Executing smart Contracts ("PArSEC")
+We built a system that is capable of performing parallel executions of virtual machines for generic smart contract support.
```
Comment (AlexRamRam) [inline README.md]: How about: `We built a system with a generic virtual machine layer that is capable of performing parallel executions of smart contracts`. b/c programs/smart contracts are executed, not virtual machines

File: src/parsec/runtime_locking_shard/replicated_shard_interface.hpp [hunk 3]
```diff
@@ -35,7 +35,7 @@ namespace cbdc::threepc::runtime_locking_shard {
             broker_id_type m_broker_id{};
             /// State updated to apply after commit.
             state_update_type m_state_update;
-            /// State of the ticket within the 3PC protocol.
+            /// State of the ticket within the PARSEC protocol.
```
Comment (AlexRamRam) [inline src/parsec/runtime_locking_shard/replicated_shard_interface.hpp]: lowercase r for consistency?

File: tests/unit/parsec/util.hpp [hunk 4]
```diff
@@ -6,12 +6,12 @@
 #ifndef TEST_UNIT_THREEPC_UTIL_INC_
 #define TEST_UNIT_THREEPC_UTIL_INC_
```
Comment (AlexRamRam) [inline tests/unit/parsec/util.hpp]: THREEPC --> PARSEC

**Review by AlexRamRam — CHANGES_REQUESTED** [review]
Noticed only a few very minor misses. Looks Great

**Review by AlexRamRam — APPROVED** [review]

**Comment by HalosGhost** [comment]
Some small changes made ad-hoc; but now included in #213

**Pull request closed without merging**